app/assets/v2/js/vue-components.js

```diff
@@ -9,10 +9,16 @@ Vue.mixin({
     };
   },
   methods: {
-    chatWindow: function(handle) {
+    chatWindow: function(channel, dm) {
+      dm = dm || !channel;
+      channel = channel || 'town-square';
+
       let vm = this;
+      const hackathonTeamSlug = 'hackathons';
+      const gitcoinTeamSlug = 'gitcoin';
+      const isHackathon = (document.hackathon_id !== null);
 
-      const url = handle ? `${vm.chatURL}/hackathons/messages/@${handle}` : `${vm.chatURL}/`;
+      const url = channel ? `${vm.chatURL}/${isHackathon ? hackathonTeamSlug : gitcoinTeamSlug}/${dm ? 'messages' : 'channel'}/@${channel}` : `${vm.chatURL}/`;
```

Comment:
is this taking out the functionality to send DM to users? chatWindow exist in others part of the site already and have a diff functionality. 
If I understand here if channel is not defined the link to open the chat will be without the user handle .
Can we use also better params names? not sure "dm" is for the handle?